addons/sale/migrations/9.0.1.0/openupgrade_analysis_work.txt

```diff
@@ -0,0 +1,199 @@
+#Nothing To Do : ORM will change the name automatically.
+
+sale         / sale.order               / invoice_ids (many2many)       : DEL relation: account.invoice
+#Nothing to do : Functionality removed now no longer needed.
+
+sale         / sale.order               / message_follower_ids (many2many): not a function anymore
+sale         / sale.order               / message_follower_ids (many2many): relation is now 'mail.followers' ('res.partner')
+sale         / sale.order               / message_follower_ids (many2many): type is now 'one2many' ('many2many')
+# TODO : ???  # Concerns module Mail (has to be done in another migration script specifically for that module) ---> Nothing to do
+
+sale         / sale.order               / order_policy (selection)      : DEL required: required, selection_keys: ['manual'], req_default: manual
+#Nothing To Do : Functionality removed now no longer needed.
+
+sale         / sale.order               / payment_term (many2one)       : was renamed to payment_term_id [nothing to to]
+#Nothing To Do : ORM will change the name automatically.
+
+sale         / sale.order               / section_id (many2one)         : DEL relation: crm.case.section
+sale         / sale.order               / team_id (many2one)            : NEW relation: crm.team
+#TODO : section_id is now team_id
+#Done : renamed the column to new column team_id.
+
+sale         / sale.order               / state (selection)             : selection_keys is now '['cancel', 'done', 'draft', 'sale', 'sent']' ('['cancel', 'done', 'draft', 'invoice_except', 'manual', 'progress', 'sent', 'shipping_except', 'waiting_date']')
+TODO : To map the state values as per new values. 
+#Done : Mapped the values as per new state values
+
+sale         / sale.order               / validity_date (date)          : NEW
+#Nothing To Do : New Functionality added.
+
+sale         / sale.order.line          / address_allotment_id (many2one): DEL relation: res.partner
+#Nothing To Do : Functionality removed now no longer needed.
+
+sale         / sale.order.line          / delay (float)                 : was renamed to customer_lead [nothing to to]
+#Nothing To Do : ORM will change the name automatically.
+
+sale         / sale.order.line          / product_id (many2one)         : now required
+#TODO : Maybe To create product from description. (need more analysis on this) 
+#Done : Created Service type product and assigned sale.order.line where product_id is null
+
+sale         / sale.order.line          / product_uos (many2one)        : DEL relation: product.uom
+#TODO : Moved to product_uos module (so saved it for later use.)
+
+sale         / sale.order.line          / product_uos_qty (float)       : DEL 
+#TODO : Moved to product_uos module (so saved it for later use.)
+
+sale         / sale.order.line          / qty_delivered (float)         : NEW 
+#Nothing To Do : New Functionality added.
```

Comment:
Migration needed. Unless you can show that this field is meaningless in existing orders after the migration, we need to try and retrieve the value for every line on every open sale order.